tools/src/podio-dump-tool.cpp

```diff
@@ -0,0 +1,267 @@
+}
+
+ParsedArgs parseArgs(std::vector<std::string> argv) {
+  // find help or version
+  if (const auto it = findFlags(argv, "-h", "--help", "--version"); it != argv.end()) {
+    if (*it == "--version") {
+      fmt::print("podio {}\n", podio::version::build_version);
+    } else {
+      fmt::print("{}\n{}", usageMsg, helpMsg);
+    }
+    std::exit(0);
+  }
+
+  ParsedArgs args;
+  // detailed flag
+  if (const auto it = findFlags(argv, "-d", "--detailed"); it != argv.end()) {
+    args.detailed = true;
+    argv.erase(it);
+  }
+  // category
+  if (const auto it = findFlags(argv, "-c", "--category"); it != argv.end()) {
+    args.category = getArgumentValueOrExit(argv, it);
+    argv.erase(it, it + 2);
+  }
+  // event range
+  if (const auto it = findFlags(argv, "-e", "--entries"); it != argv.end()) {
+    args.events = parseEventRange(*(it + 1));
+    argv.erase(it, it + 2);
+  }
+  // dump-edm
+  if (const auto it = findFlags(argv, "--dump-edm"); it != argv.end()) {
+    args.dumpEDM = getArgumentValueOrExit(argv, it);
+    argv.erase(it, it + 2);
+  }
+
+  if (argv.size() != 1) {
+    printUsageAndExit();
+  }
+  args.inputFile = argv[0];
+
+  return args;
+}
+
+template <typename T>
+std::string getTypeString() {
+  if constexpr (std::is_same_v<T, int>) {
+    return "int";
+  } else if constexpr (std::is_same_v<T, float>) {
+    return "float";
+  } else if constexpr (std::is_same_v<T, double>) {
+    return "double";
+  } else if constexpr (std::is_same_v<T, std::string>) {
+    return "std::string";
+  }
+
+  return "unknown";
+}
+
+template <typename T>
+void getParameterOverview(const podio::Frame& frame, std::vector<std::tuple<std::string, std::string, size_t>>& rows) {
+  const auto typeString = getTypeString<T>();
+  for (const auto& parKey : podio::utils::sortAlphabeticaly(frame.getParameterKeys<T>())) {
+    rows.emplace_back(parKey, typeString, frame.getParameter<std::vector<T>>(parKey)->size());
+  }
+}
+
+void printFrameOverview(const podio::Frame& frame) {
+  fmt::print("Collections:\n");
+  const auto collNames = frame.getAvailableCollections();
+
+  std::vector<std::tuple<std::string, std::string_view, size_t, std::string>> rows;
+  rows.reserve(collNames.size());
+
+  for (const auto& name : podio::utils::sortAlphabeticaly(collNames)) {
+    const auto coll = frame.get(name);
+    rows.emplace_back(name, coll->getValueTypeName(), coll->size(), fmt::format("{:0>8x}", coll->getID()));
+  }
+  printTable(rows, {"Name", "ValueType", "Size", "ID"});
+
+  fmt::print("\nParameters:\n");
+  std::vector<std::tuple<std::string, std::string, size_t>> paramRows{};
+  getParameterOverview<int>(frame, paramRows);
+  getParameterOverview<float>(frame, paramRows);
+  getParameterOverview<double>(frame, paramRows);
+  getParameterOverview<std::string>(frame, paramRows);
+
+  printTable(paramRows, {"Name", "Type", "Elements"});
+}
+
+template <typename... Args>
+void print_flush(fmt::format_string<Args...> fmtstr, Args&&... args) {
```

Comment:
```suggestion
void print_flush(const fmt::format_string<Args...>& fmtstr, Args&&... args) {
```

Reply:
The whole function has been removed, as it is no longer necessary.